File: tkpdsession/src/main/java/com/tokopedia/otp/phoneverification/view/fragment/ReferralPhoneNumberVerificationFragment.java
package com.tokopedia.otp.phoneverification.view.fragment;

import android.app.Activity;
import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import com.tokopedia.core.analytics.AppEventTracking;
import com.tokopedia.core.analytics.UnifyTracking;
import com.tokopedia.core.app.BasePresenterFragment;
import com.tokopedia.core.util.CustomPhoneNumberUtil;
import com.tokopedia.core.util.SessionHandler;
import com.tokopedia.otp.phoneverification.view.activity.ChangePhoneNumberActivity;
import com.tokopedia.session.R;

/**
 * Created by ashwanityagi on 12/12/17.
 */

public class ReferralPhoneNumberVerificationFragment extends BasePresenterFragment {

    public interface ReferralPhoneNumberVerificationFragmentListener {
        void onSkipVerification();

        void onClickVerification(String phoneNumber);
    }


    EditText tvPhoneNumber;
    TextView btnActivation;
    private ReferralPhoneNumberVerificationFragmentListener listener;
    private SessionHandler sessionHandler;

    public static ReferralPhoneNumberVerificationFragment newInstance() {
        ReferralPhoneNumberVerificationFragment fragment = new ReferralPhoneNumberVerificationFragment();
        return fragment;
    }

    public static ReferralPhoneNumberVerificationFragment createInstance(ReferralPhoneNumberVerificationFragmentListener listener) {
        ReferralPhoneNumberVerificationFragment fragment = new ReferralPhoneNumberVerificationFragment();
        fragment.setReferralPhoneVerificationListener(listener);
        return fragment;
    }

    public void setReferralPhoneVerificationListener(ReferralPhoneNumberVerificationFragmentListener listener) {
        this.listener = listener;
    }

    public ReferralPhoneNumberVerificationFragmentListener getListener() {
        return listener;
    }


    @Override
    public void onSaveState(Bundle state) {

    }

    @Override
    public void onRestoreState(Bundle savedState) {

    }

    @Override
    protected void onFirstTimeLaunched() {
        if(sessionHandler == null){
            sessionHandler = new SessionHandler(getActivity());
        }
        tvPhoneNumber.setText(CustomPhoneNumberUtil.transform(
                sessionHandler.getPhoneNumber()));
    }

    @Override
    protected boolean isRetainInstance() {
        return false;
    }

    @Override
    protected void initialPresenter() {

    }

    @Override
    protected void initialListener(Activity activity) {

    }

    @Override
    protected void setupArguments(Bundle arguments) {

    }

    @Override
    protected int getFragmentLayout() {
        return R.layout.fragment_referral_phone_number_verification;
    }


    @Override
    protected void initView(View view) {
        if(sessionHandler == null){
            sessionHandler = new SessionHandler(getActivity());
        }
        tvPhoneNumber = (EditText) view.findViewById(R.id.tv_phone_number);
        tvPhoneNumber.setText(CustomPhoneNumberUtil.transform(
                sessionHandler.getPhoneNumber()));
        btnActivation = (TextView) view.findViewById(R.id.btn_activation);
        setViewListener();
        tvPhoneNumber.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                startActivityForResult(
                        ChangePhoneNumberActivity.getChangePhoneNumberIntent(
                                getActivity(),
                                tvPhoneNumber.getText().toString()),
                        ChangePhoneNumberFragment.ACTION_CHANGE_PHONE_NUMBER);
            }
        });
    }

    @Override
    protected void setViewListener() {
        btnActivation.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                if (listener != null) {
                    UnifyTracking.eventReferralAndShare(AppEventTracking.Action.CLICK_VERIFY_NUMBER,tvPhoneNumber.getText().toString().replace("-", ""));

                    listener.onClickVerification(tvPhoneNumber.getText().toString());
                }
            }
        });

    }

    @Override
    public void onActivityResult(int requestCode, int resultCode, Intent data) {
        if (requestCode == ChangePhoneNumberFragment.ACTION_CHANGE_PHONE_NUMBER &&
                resultCode == Activity.RESULT_OK) {
            tvPhoneNumber.setText(data.getStringExtra(ChangePhoneNumberFragment.EXTRA_PHONE_NUMBER));
        }
        super.onActivityResult(requestCode, resultCode, data);
    }


    @Override
    protected void initialVar() {

    }

    @Override
    protected void setActionVar() {

    }

    @Override
    protected boolean getOptionsMenuEnable() {
        return false;
    }

}

File: features/im/referral/src/main/java/com/tokopedia/referral/view/fragment/ReferralPhoneNumberVerificationFragment.kt
package com.tokopedia.referral.view.fragment

import com.tokopedia.imageassets.TokopediaImageUrl

import android.app.Activity
import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.EditText
import android.widget.ImageView
import android.widget.TextView
import com.tokopedia.abstraction.base.view.fragment.BaseDaggerFragment
import com.tokopedia.media.loader.loadImageWithError
import com.tokopedia.applink.RouteManager
import com.tokopedia.applink.internal.ApplinkConstInternalGlobal
import com.tokopedia.applink.internal.ApplinkConstInternalUserPlatform
import com.tokopedia.referral.R
import com.tokopedia.referral.analytics.ReferralPhoneVerificationAnalytics
import com.tokopedia.user.session.UserSession
import com.tokopedia.utils.phonenumber.PhoneNumberUtil.transform


/**
 */
class ReferralPhoneNumberVerificationFragment : BaseDaggerFragment() {
    override fun initInjector() {
    }

    var tvPhoneNumber: EditText? = null
    var btnActivation: TextView? = null
    private var ivTokocash: ImageView? = null
    private var analytics: ReferralPhoneVerificationAnalytics? = null
    lateinit var userSession: UserSession


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        analytics = ReferralPhoneVerificationAnalytics.createInstance()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_referral_phone_number_verification,
                container, false)
        initView(view)
        return view
    }

    override fun getScreenName(): String {
        return ""
    }

    private fun initView(view: View) {
        userSession = UserSession(activity)
        tvPhoneNumber = view.findViewById<View>(R.id.tv_phone_number) as EditText
        tvPhoneNumber?.setText(transform(userSession.phoneNumber))
        btnActivation = view.findViewById<View>(R.id.btn_activation) as TextView
        ivTokocash = view.findViewById<View>(R.id.img_app_share) as ImageView
        ivTokocash?.loadImageWithError(URL_TOKOCASH_SHARE, com.tokopedia.design.R.drawable.loading_page)
        setViewListener()
    }

    private fun setViewListener() {
        btnActivation?.setOnClickListener {
            val rawPhone = tvPhoneNumber?.text.toString().replace("-", "")
            analytics?.eventReferralAndShare(
                    ReferralPhoneVerificationAnalytics.Action.CLICK_VERIFY_NUMBER,
                    tvPhoneNumber?.text.toString().replace("-", ""))
            val intent = RouteManager.getIntent(activity, ApplinkConstInternalUserPlatform.ADD_PHONE_WITH, rawPhone)
            intent.addFlags(Intent.FLAG_ACTIVITY_FORWARD_RESULT)
            startActivity(intent)
            activity?.finish()
        }
    }

    companion object {
        fun newInstance(): ReferralPhoneNumberVerificationFragment {
            return ReferralPhoneNumberVerificationFragment()
        }
        const val EXTRA_PHONE = "phone"
        const val URL_TOKOCASH_SHARE = TokopediaImageUrl.URL_TOKOCASH_SHARE
    }
}
